### ISSUE TITLE
macOS: 'fsync' fails write to SMB share

### ISSUE THREAD

**Issue by U47** [neovim/neovim#6725]
- `nvim --version`:
```
NVIM v0.2.1-151-gedfe0980
Build type: RelWithDebInfo
Compilation: /usr/local/Homebrew/Library/Homebrew/shims/super/clang -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_MSGPACK_HAS_FLOAT32 -O2 -g -DDISABLE_LOG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -I/tmp/neovim-20170511-47423-9x0cl0/build/config -I/tmp/neovim-20170511-47423-9x0cl0/src -I/usr/local/include -I/usr/local/include -I/usr/local/include -I/usr/local/include -I/usr/local/include -I/usr/local/include -I/usr/local/opt/gettext/include -I/usr/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/usr/include -I/tmp/neovim-20170511-47423-9x0cl0/build/src/nvim/auto -I/tmp/neovim-20170511-47423-9x0cl0/build/include
Compiled by xxxx@xxxx.local

Optional features included (+) or not (-): +acl   +iconv    +jemalloc +tui
For differences from Vim, see :help vim-differences

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/Cellar/neovim/HEAD-edfe098/share/nvim"
```

- Vim (version: ) behaves differently?
Seems to only have started with v0.2.

- Operating system/version:
macOS Sierra 10.12.4

- Terminal name/version:
iterm2 3.0.15, tmux 2.3

- `$TERM`:
screen-256color

### Actual behaviour
`E667: Fsync failed: operation not supported on socket`

### Expected behaviour
Buffer is written

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
:e /Volumes/smbshare/path/to/file
:w
```

**Comment by U47** [comment]
Vanilla VIM has no issue writing the buffer, fwiw.

**Comment by U47** [comment]
Works with v0.1.5.

**Comment by teto** [comment]
Have you tried setting nofsync ?
(thought of the option because of https://github.com/neovim/neovim/issues/6437)

**Comment by pratz** [comment]
Facing the same issue, however, setting `set nofsync` seems to be a workaround.

**Comment by iate49monkeys** [comment]
I am also experiencing the same issue. I can't write to a SMB share using Neovim (fsync failed error) but can with standard vim.

*    OS: macos 10.12.5
*    Neovim Version: 0.2.0 (installed using homebrew)

SMB Share information (from `smbutil statshares -a`)


SHARE                       | ATTRIBUTE TYPE               | VALUE
-------------------- | -------------------------- | ---------
XXX | . | .
.  | SERVER_NAME  | XXX
.  | USER_ID | XXX
.  | SMB_NEGOTIATE  | SMBV_NEG_SMB1_ENABLED
.  | SMB_VERSION  | SMB_1
.  | SMB_SHARE_TYPE  | UNKNOWN
.  | EXTENDED_SECURITY_SUPPORTED   | TRUE
.  | LARGE_FILE_SUPPORTED  | TRUE
.  | CLIENT_REQUIRES_SIGNING  | TRUE

**Comment by justinmk** [comment]
set nofsync

**Comment by justinmk** [comment]
@ZyX-I Would it be worth retrying a write without fsync, if it fails with fsync enabled?

**Comment by ZyX-I** [comment]
@U47 And there still is a question whether Vim behaves the same (with `set fsync`, of course). Maybe it already has some detection we could borrow.

**Comment by johnzeng** [comment]
+1 for this bug.And `set nofsync` fix problem. But it is still a bug...

**Comment by gilesorr** [comment]
Same problem: MacOS (10.12.6), just did 'brew upgrade neovim' and got v0.2.0.  Fails on attempts to write to SMB shares.  ':set nofsync' appears to fix the problem.  It wasn't a problem with the older version of nvim.

**Comment by gilesorr** [comment]
This went away for a long while - I've been using a brew installed version 0.2.2 since at least November, and it was fixed before that.  But yesterday through brew I got 0.2.2_1 and suddenly the problem is back.  Again, ":set nofsync" fixes the problem, but better I shouldn't have to use that option.  Mac 10.13.3.  Thanks.  (I'm not sure if this is a brew issue rather than a Neovim issue?  Seems like a regression though.)
